### PR TITLE
Dependencies setup

### PR DESCRIPTION
**Pull request by LittleJennie**

### CODE REVIEW

**Review by cminho921 — COMMENTED** [review]
reviewed env-setup

File: client/components/app.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
Comment (RamKhalsa) [inline client/components/app.jsx:1]: @LittleJennie 
@cminho921 

Again, a bit draconian as we know that this is going to be stateful eventually, but until it is, components should be functional stateless if the constructor and render methods are empty!

Also, remember the new line at the end of the file!

File: client/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './components/app.jsx';
+
+ReactDOM.render(
+    <App />,
+    document.getElementById('app')
+);
```
Comment (RamKhalsa) [inline client/index.js:8]: @cminho921 
@LittleJennie 

New line needed here!

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "reservations",
+  "version": "1.0.0",
+  "description": "reservation module for mock up Open Table page",
```
Comment (RamKhalsa) [inline package.json:4]: @LittleJennie 
@cminho921 

This should not mention Open Table--remember that this organization will stand publicly, and is about demonstrating your ability to build a component that meets a need or has certain functionality vs. your ability to clone something.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "main": "./server/index.js",
+  "scripts": {
+    "test": "node_modules/jest/bin/jest.js --watch",
```
Comment (RamKhalsa) [inline package.json:7]: @LittleJennie 

You may want to look into NPX and running commands via files from your node modules here, if you're interested in another way to do this!

File: package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "start": "nodemon -w ./server/index.js",
```
Comment (RamKhalsa) [inline package.json:8]: @LittleJennie 
@cminho921 

From Nodemon's documentation:

By default nodemon monitors the current working directory. If you want to take control of that option, use the --watch option to add specific paths:

nodemon --watch app --watch libs app/server.js

Now nodemon will only restart if there are changes in the ./app or ./libs directory. By default nodemon will traverse sub-directories, so there's no need in explicitly including sub-directories.

Is this a purposeful change to only watch the server/index file for changes vs. any changes to any .js files in your working directory?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  "devDependencies": {
+    "@babel/core": "^7.4.5",
+    "@babel/preset-env": "^7.4.5",
+    "@babel/preset-react": "^7.0.0",
+    "babel-loader": "^8.0.6",
+    "jest": "^24.8.0",
+    "morgan": "^1.9.1",
+    "path": "^0.12.7",
+    "puppeteer": "^1.18.1",
+    "react": "^16.8.6",
```
Comment (RamKhalsa) [inline package.json:30]: @LittleJennie 
@cminho921 

Some of these dev dependencies are actually dependencies--when deciding between them, think of what a user might need to have on their computer to run your server, database, and client.

e.g., your dev dependencies will test your code for you and ensure it works, lint the code to ensure it is clean and formatted, bundle it and prepare it for distribution, but then ....what dependencies does the user have?

File: public/index.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf8">
+        <title>Mock Up Open Table Reservation Module</title>
```
Comment (RamKhalsa) [inline public/index.html:5]: @LittleJennie 
@cminho921 

Would change this document title as well!

File: test/sum.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+function sum(a, b) {
+    return a + b;
+  }
+module.exports = sum;
```
Comment (RamKhalsa) [inline test/sum.js:4]: @LittleJennie 
@cminho921 

Two comments here:

1. Would recommend not committing and including any files that you have to test functionality (e.g., if Sum is a part of setting Jest up, but has nothing to do with your app, I would add them to your gitignore in the future).

2. This file and the next are missing newlines to be aligned (no pun intended) with the Airbnb style guide!

**Review by RamKhalsa — COMMENTED** [review]
@LittleJennie 
@cminho921 

Added some comments to be reconciled, understanding that this has been merged already! Nice job setting up the Jest environment.

File: client/components/app.jsx (hunk 1 above)
Comment (LittleJennie) [reply client/components/app.jsx:1]: Thanks for pointing this out Ram! I didn't set up eslint until my 2nd or 3rd PR. So the this PR is not following Airbnb style guide at all. I was able to fix the styles in the later PR.

File: package.json (hunk 5 above)
Comment (LittleJennie) [reply package.json:8]: I didn't quite understand the 'watch' command here at first. The '-w' command is later removed in the 4th PR.

File: package.json (hunk 6 above)
Comment (LittleJennie) [reply package.json:30]: 'path', 'react' and 'react-dom' are moved out of dev-dependencies in later PR.

File: package.json (hunk 3 above)
Comment (LittleJennie) [reply package.json:4]: Thank you for point this out! Description will be updated in the next PR.

File: public/index.html (hunk 7 above)
Comment (LittleJennie) [reply public/index.html:5]: Thank you for point this out! HTML title will be updated in the next PR.

**Pull request merged**